x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/tasks/custom_response_actions_rule.ts

```diff
@@ -0,0 +1,142 @@
+  const taskName = 'Security Solution Response Actions Rules Telemetry';
+  const taskType = 'security:telemetry-response-actions-rules';
+  return {
+    type: taskType,
+    title: taskName,
+    interval: '24h',
+    timeout: '10m',
+    version: '1.0.0',
+    runTask: async (
+      taskId: string,
+      logger: Logger,
+      receiver: ITelemetryReceiver,
+      sender: ITelemetryEventsSender,
+      taskMetricsService: ITaskMetricsService,
+      taskExecutionPeriod: TaskExecutionPeriod
+    ) => {
+      const mdc = { task_id: taskId, task_execution_period: taskExecutionPeriod };
+      const log = newTelemetryLogger(logger.get('response_actions_rules'), mdc);
+      const usageCollector = sender.getTelemetryUsageCluster();
+      const usageLabelEndpointPrefix: string[] = [
+        'security_telemetry',
+        'endpoint-response-actions-rules',
+      ];
+      const usageLabelOsqueryPrefix: string[] = [
+        'security_telemetry',
+        'osquery-response-actions-rules',
+      ];
+      const trace = taskMetricsService.start(taskType);
+
+      log.l('Running response actions rules telemetry task');
+
+      try {
+        const [clusterInfoPromise, licenseInfoPromise] = await Promise.allSettled([
+          receiver.fetchClusterInfo(),
+          receiver.fetchLicenseInfo(),
+        ]);
+
+        const clusterInfo = safeValue(clusterInfoPromise);
+        const licenseInfo = safeValue(licenseInfoPromise);
+
+        const {
+          body: { aggregations },
+        } = (await receiver.fetchResponseActionsRules()) ?? {};
+
+        if (!aggregations || !aggregations.actionTypes) {
+          log.debug('no custom response action rules found');
+          await taskMetricsService.end(trace);
+          return 0;
+        }
+
+        const responseActionRules = (
+          aggregations as unknown as ResponseActionsRuleResponseAggregations
+        ).actionTypes.buckets.reduce<ResponseActionRules>((acc, agg) => {
+          if (agg.key === '.endpoint') {
+            acc.endpoint = agg.rulesInfo.buckets.map((rule) => rule.key);
+          } else if (agg.key === '.osquery') {
+            acc.osquery = agg.rulesInfo.buckets.map((rule) => rule.key);
+          }
+          return acc;
+        }, {} as ResponseActionRules);
+
+        const shouldNotProcessTelemetry =
+          responseActionRules.endpoint === undefined ||
+          responseActionRules.osquery === undefined ||
+          responseActionRules.endpoint.length === 0 ||
+          responseActionRules.osquery.length === 0;
+
+        if (shouldNotProcessTelemetry) {
+          return 0;
+        }
+
+        const responseActionsRulesTelemetryData = responseActionsCustomRuleTelemetryData(
+          responseActionRules,
+          clusterInfo,
+          licenseInfo
+        );
+
+        log.l('Custom response actions rules data', {
+          data: JSON.stringify(responseActionsRulesTelemetryData),
```

Comment:
is it fine to log the full json? aren't rule ids or action_type_id sensitive data?

Reply:
I'm not sure rule uuids are sensitive. I would exclude it if that is the case. This is not collecting action_type_id though. I only need a count of distinct rule ids that have osquery or endpoint actions configured.